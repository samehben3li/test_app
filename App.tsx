import { useState } from "react";
import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import HomeScreen from "./screens/home";
import LoginScreen from "./screens/login";

const Stack = createNativeStackNavigator();

export default function App() {
  const [isSignedIn, setIsSignedIn] = useState(false);
  return (
    <NavigationContainer>
      <Stack.Navigator screenOptions={{ headerShown: false }}>
        {isSignedIn ? (
          <>
            <Stack.Screen
              name="home"
              component={HomeScreen}
              initialParams={{ setIsSignedIn: setIsSignedIn }}
            />
          </>
        ) : (
          <>
            <Stack.Screen
              name="login"
              component={LoginScreen}
              initialParams={{ setIsSignedIn: setIsSignedIn }}
            />
          </>
        )}
      </Stack.Navigator>
    </NavigationContainer>
  );
}
