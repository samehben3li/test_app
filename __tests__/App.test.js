import React from "react";
import renderer from "react-test-renderer";
import App from "../App";

describe("App", () => {
  it("renders without crashing", () => {
    const rendered = renderer.create(<App />).toJSON();
    expect(rendered).toBeTruthy();
  });
});
// test("home snapshot", () => {
//   const snap = renderer.create(<App />).toJSON();
//   expect(snap).toMatchSnapshot();
// });

// it("App test against snapshot", () => {
//   const tree = renderer.create(<App />).toJSON();
//   expect(tree).toMatchSnapshot();
// });
