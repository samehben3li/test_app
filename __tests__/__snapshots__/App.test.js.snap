// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Loading...
  </Text>
</View>
`;
